extensions/azurecore/src/azureResource/services/tenantService.ts

```diff
@@ -2,29 +2,17 @@
  *  Copyright (c) Microsoft Corporation. All rights reserved.
  *  Licensed under the Source EULA. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
-
-import * as request from 'request';
+import { SubscriptionClient } from '@azure/arm-subscriptions';
 
 import { azureResource } from '../azure-resource';
 import { IAzureResourceTenantService } from '../interfaces';
+import { Account } from 'azdata';
 
 export class AzureResourceTenantService implements IAzureResourceTenantService {
-	public async getTenantId(subscription: azureResource.AzureResourceSubscription): Promise<string> {
-		const requestPromisified = new Promise<string>((resolve, reject) => {
-			const url = `https://management.azure.com/subscriptions/${subscription.id}?api-version=2014-04-01`;
-			request(url, function (error, response, body) {
-				if (response.statusCode === 401) {
-					const tenantIdRegEx = /authorization_uri="https:\/\/login\.windows\.net\/([0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12})"/;
-					const teantIdString = response.headers['www-authenticate'];
-					if (tenantIdRegEx.test(teantIdString)) {
-						resolve(tenantIdRegEx.exec(teantIdString)[1]);
-					} else {
-						reject();
-					}
-				}
-			});
-		});
+	public async getTenantId(subscription: azureResource.AzureResourceSubscription, account: Account, credentials: any): Promise<string> {
```

Comment:
You can't use `msRest.ServiceClientCredentials` as the type of credentials here? That's what you have in the interface.

Also the param name here doesn't match - it's credential in the interface def.

Reply:
I honestly have no idea why this doesn't work as the type... So I changed it again to get the error:

```
[10:35:32] Error: C:/Users/amomidi/Documents/azuredatastudio-insiders/extensions/azurecore/src/azureResource/services/tenantService.ts(14,44): Argument of type 'import("C:/Users/amomidi/Documents/azuredatastudio-insiders/extensions/azurecore/node_modules/@azure/ms-rest-js/es/lib/credentials/serviceClientCredentials").ServiceClientCredentials' is not assignable to parameter of type 'import("C:/Users/amomidi/Documents/azuredatastudio-insiders/extensions/azurecore/node_modules/@azure/arm-subscriptions/node_modules/@azure/ms-rest-js/es/lib/credentials/serviceClientCredentials").ServiceClientCredentials'.
  Types of property 'signRequest' are incompatible.
    Type '(webResource: import("C:/Users/amomidi/Documents/azuredatastudio-insiders/extensions/azurecore/node_modules/@azure/ms-rest-js/es/lib/webResource").WebResource) => Promise<import("C:/Users/amomidi/Documents/azuredatastudio-insiders/extensions/azurecore/node_modules/@azure/ms-rest-js/es/lib/webResource").WebResource>' is not assignable to type '(webResource: import("C:/Users/amomidi/Documents/azuredatastudio-insiders/extensions/azurecore/node_modules/@azure/arm-subscriptions/node_modules/@azure/ms-rest-js/es/lib/webResource").WebResource) => Promise<...>'.
      Types of parameters 'webResource' and 'webResource' are incompatible.
        Type 'import("C:/Users/amomidi/Documents/azuredatastudio-insiders/extensions/azurecore/node_modules/@azure/arm-subscriptions/node_modules/@azure/ms-rest-js/es/lib/webResource").WebResource' is not assignable to type 'import("C:/Users/amomidi/Documents/azuredatastudio-insiders/extensions/azurecore/node_modules/@azure/ms-rest-js/es/lib/webResource").WebResource'.
          Types of property 'headers' are incompatible.
            Type 'import("C:/Users/amomidi/Documents/azuredatastudio-insiders/extensions/azurecore/node_modules/@azure/arm-subscriptions/node_modules/@azure/ms-rest-js/es/lib/httpHeaders").HttpHeaders' is not assignable to type 'import("C:/Users/amomidi/Documents/azuredatastudio-insiders/extensions/azurecore/node_modules/@azure/ms-rest-js/es/lib/httpHeaders").HttpHeaders'.
              Types have separate declarations of a private property '_headersMap'.
```

Reply:
It seems like they messed up their typings?